test/jttt/Core/UI/CommandLineUITest.java

```diff
@@ -1,284 +1,363 @@
+        CommandLineUI cli = new CommandLineUI(game, inputStream, printStream);
+        assertEquals(1, cli.requestGameType());
+        String expected = cli.GAME_TYPE_REQUEST;
+        assertThat(output.toString(), containsString(expected));
     }
 
     @Test
-    public void checkResultIsAWin() {
-        List<Integer> initialState = new ArrayList<>(Arrays.asList(
-                1, 2, 3,
-                4, 5, 6,
-                7, 8, 9));
-        fakeUI = generateFakeUI(initialState, 3, 1);
-        fakeUI.start();
-        assertEquals(true, fakeUI.getGame().isGameOver());
-        assertEquals(false, fakeUI.isADraw());
-        assertEquals(Mark.X, fakeUI.getWinner());
+    public void requestNextMoveCalled() {
+        InputStream inputStream = new ByteArrayInputStream("9\n".getBytes());
+        CommandLineUI cli = new CommandLineUI(game, inputStream, printStream);
+        assertEquals(9, cli.requestNextPosition());
+        String expected = cli.POSITION_REQUEST;
+        assertThat(output.toString(), containsString(expected));
     }
 
     @Test
-    public void humanVsComputer() {
-        List<Integer> initialState = new ArrayList<>(Arrays.asList(
-                2, 1, 4,
-                5, 6, 3,
-                7, 8, 9));
-        fakeUI = generateFakeUI(initialState, 3, 2);
-        fakeUI.start();
-        assertEquals(HumanPlayer.class, fakeUI.getGame().getPlayer(Mark.X).getClass());
-        assertEquals(ComputerPlayer.class, fakeUI.getGame().getPlayer(Mark.O).getClass());
-    }
+    public void resultDisplayed() {
+        byte[] buf = setupGame(GameType.HVH, 3, new int[]{3, 1, 2, 4, 6, 5, 7, 9, 8}, Replay.QUIT);
+        InputStream inputStream = new ByteArrayInputStream(buf);
+        CommandLineUI cli = new CommandLineUI(game, inputStream, printStream);
+        cli.start();
+        String expected = cli.REPLAY_REQUEST;
+        assertThat(output.toString(), containsString(expected));
 
-    @Test
-    public void computerVsHuman() {
-        List<Integer> initialState = new ArrayList<>(Arrays.asList(
-                2, 1, 4,
-                5, 6, 3,
-                7, 8, 9));
-        fakeUI = generateFakeUI(initialState, 3, 3);
-        fakeUI.start();
-        assertEquals(ComputerPlayer.class, fakeUI.getGame().getPlayer(Mark.X).getClass());
-        assertEquals(HumanPlayer.class, fakeUI.getGame().getPlayer(Mark.O).getClass());
     }
 
-    @Test
-    public void duplicateMoveRepeatsMoveRequestTillValid() {
-        fakeUI = generateFakeUI(Arrays.asList(
-                1, 2, 2,
-                4, 7, 3,
-                8, 9, 5), 3, 1);
-        fakeUI.start();
-        assertEquals("" +
-                        "[X][O][X]\n" +
-                        "[X][O][6]\n" +
-                        "[O][O][X]\n",
-                fakeUI.displayBoard());
+    private byte[] setupGame(GameType gameType, int dimension, int[] humanMoves, Replay choice) {
```

Comment:
This a very big and fairly complex method for what seems to be a simple test. If you were to inline this in the only test  that uses it, it would look very long. How can this be simplified?